### PR TITLE
SONAR-6247 Add createdSince parameter to search issues by lifetime

### PR DESCRIPTION
**Pull request by jblievremont**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/issue/IssueQueryService.java [hunk 1]
```diff
@@ -135,6 +137,20 @@ public IssueQuery createFromMap(Map<String, Object> params) {
     }
   }
 
+  private Date buildCreatedAfter(@Nullable Date createdAfter, @Nullable String createdSince) {
```
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/issue/IssueQueryService.java]: I would use Preconditions.checkArgument(createdAfter == null || createdSince == null, "createdAfter and createdSince cannot be set simultaneously") at the beginning of the method

File: server/sonar-server/src/test/java/org/sonar/server/issue/IssueQueryServiceTest.java [hunk 2]
```diff
@@ -434,4 +435,41 @@ public void should_search_on_test_file() throws Exception {
...
+  }
+
+  @Test
+  public void set_created_after_from_created_since() {
+    Date now = DateUtils.parseDateTime("2013-07-25T07:35:00+0100");
+    when(system.now()).thenReturn(now.getTime());
+    Map<String, Object> map = newHashMap();
+
+    map.put("createdSince", "1d");
+    assertThat(issueQueryService.createFromMap(map).createdAfter()).isEqualTo(DateUtils.parseDateTime("2013-07-24T07:35:00+0100"));
```
Comment (teryk) [inline server/sonar-server/src/test/java/org/sonar/server/issue/IssueQueryServiceTest.java]: Don't you think one test is enough, the rest is testing the ISOStandardPeriod formatting

File: server/sonar-server/src/test/java/org/sonar/server/issue/ws/SearchActionMediumTest.java [hunk 3]
```diff
@@ -88,7 +88,7 @@ public void define_action() throws Exception {
     assertThat(show.isPost()).isFalse();
     assertThat(show.isInternal()).isFalse();
     assertThat(show.responseExampleAsString()).isNotEmpty();
-    assertThat(show.params()).hasSize(39);
+    assertThat(show.params()).hasSize(40);
```
Comment (teryk) [inline server/sonar-server/src/test/java/org/sonar/server/issue/ws/SearchActionMediumTest.java]: a monster ws action, 40 possible parameters, woaw … (no action needed)

File: server/sonar-server/src/test/java/org/sonar/server/issue/IssueQueryServiceTest.java (hunk 2 above)
Comment (jblievremont) [inline server/sonar-server/src/test/java/org/sonar/server/issue/IssueQueryServiceTest.java]: I see it as a kind of behavior documentation and exploratory test.

Comment (teryk) [inline server/sonar-server/src/test/java/org/sonar/server/issue/IssueQueryServiceTest.java]: I understand for the exploratory test, but it doesn't need to be kept in production test code. Concerning the behavior documentation, I would add tests in the ISOStandardPeriodTest class, if there aren't already

File: server/sonar-server/src/main/java/org/sonar/server/issue/IssueQueryService.java (hunk 1 above)
Comment (jblievremont) [inline server/sonar-server/src/main/java/org/sonar/server/issue/IssueQueryService.java]: Done.

**Comment by jblievremont** [comment]
Should be better now :)

**Pull request closed without merging**